### PR TITLE
Users

### PR DESCRIPTION
**Pull request by carmina-c**

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+@Controller("auth")
+export class AuthController {
+
+	public constructor(
+		private authService: AuthService
+	) { }
+
+	@Post("register")
+  	@UseInterceptors(ClassSerializerInterceptor)
```
Comment (danielursu) [inline src/auth/auth.controller.ts:16]: is indentation ok?

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	private register(@Body() body: RegisterDto): Promise<User|never> {
```
Comment (danielursu) [inline src/auth/auth.controller.ts]:
```suggestion
	private register(@Body() body: RegisterDTO): Promise<User|never> {
```

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		return this.authService.register(body);
+	}
+
+  	@Post("login")
+	private login(@Body() body: LoginDto): Promise<string|never> {
```
Comment (danielursu) [inline src/auth/auth.controller.ts]:
```suggestion
	private login(@Body() body: LoginDTO): Promise<string|never> {
```

File: src/auth/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@Injectable()
+export class JwtAuthGuard extends AuthGuard("jwt") implements IAuthGuard {
+
+	public handleRequest(err: unknown, user: User): any {
+		return user;
+	}
+
+	public async canActivate(context: ExecutionContext): Promise<boolean> {
+		await super.canActivate(context);
+		const user: Request = context.switchToHttp().getRequest();
```
Comment (danielursu) [inline src/auth/auth.guard.ts]:
```suggestion
		const request: Request = context.switchToHttp().getRequest();
```

File: src/auth/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+export class AuthService {
+
+	public constructor(
+		@InjectRepository(User)
+		private userRepository: Repository<User>,
+		@Inject(AuthHelper)
+		private helper: AuthHelper
+	) {}
+
+	public async register(body: RegisterDto): Promise<User | never> {
```
Comment (danielursu) [inline src/auth/auth.service.ts]:
```suggestion
	public async register(registerDTO: RegisterDTO): Promise<User | never> {
```

File: src/auth/auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { AuthHelper } from "./auth.helper";
+
+@Injectable()
+export class AuthService {
+
+	public constructor(
+		@InjectRepository(User)
+		private userRepository: Repository<User>,
+		@Inject(AuthHelper)
+		private helper: AuthHelper
```
Comment (danielursu) [inline src/auth/auth.service.ts]:
```suggestion
		private authHelper: AuthHelper
```

File: src/auth/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+		  throw new HttpException("Conflict", HttpStatus.CONFLICT);
+		}
+		user = new User();
+		user.name = name;
+		user.email = email;
+		user.password = this.helper.encodePassword(password);
+		return this.userRepository.save(user);
+	}
+
+	public async login(body: LoginDto): Promise<string | never> {
```
Comment (danielursu) [inline src/auth/auth.service.ts]:
```suggestion
	public async login(loginDTO: LoginDTO): Promise<string | never> {
```

File: src/task/task.service.ts [hunk 8]
```diff
@@ -15,23 +15,28 @@ export class TaskService {
...
-		const task: Task = await this.taskRepo.findOne({ where: { id: id } });
+		const task: Task = await this.taskRepo.findOne({
+			where: { id: id },
+			relations: ["user"]
+		});
 		return this.taskMapper.toDTO(task);
 	}
 
-	public async updateTaskById(taskDTO: TaskDTO): Promise<TaskDTO> {
+	public async updateTaskById(taskDTO): Promise<TaskDTO> {
```
Comment (danielursu) [inline src/task/task.service.ts]:
```suggestion
	public async updateTaskById(taskDTO: TaskDTO): Promise<TaskDTO> {
```

File: src/task/task.service.ts [hunk 9]
```diff
@@ -15,23 +15,28 @@ export class TaskService {
 		private taskMapper: TaskMapper
 	) { }
 
-	public addTask(taskDTO: TaskDTO): Promise<InsertResult> {
+	public addTask(taskDTO): Promise<InsertResult> {
```
Comment (danielursu) [inline src/task/task.service.ts]:
```suggestion
	public addTask(taskDTO: TaskDTO): Promise<InsertResult> {
```

File: src/user/user.mapper.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+	public toDTOs(entities: User[]): UserDTO[] {
+		const userDTOs: UserDTO[] = [];
+		for (const entity of entities) {
+			userDTOs.push(this.toDTO(entity))
+		}
+		return userDTOs;
+	}
+
+	public toEntity(dto: UserDTO): User {
```
Comment (danielursu) [inline src/user/user.mapper.ts:29]: you can delete this method if it is not used

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (carmina-c) [reply src/auth/auth.controller.ts:16]: it seems ok, 1 tab

File: src/user/user.mapper.ts (hunk 10 above)
Comment (carmina-c) [reply src/user/user.mapper.ts:29]: it's used in TaskMapper's toEntity method

File: src/task/task.service.ts (hunk 9 above)
Comment (carmina-c) [reply src/task/task.service.ts]: i get the following error when i add TaskDTO type to taskDTO parameter image

File: src/task/task.service.ts (hunk 8 above)
Comment (carmina-c) [reply src/task/task.service.ts]: same here

File: src/task/task.service.ts (hunk 9 above)
Comment (danielursu) [reply src/task/task.service.ts]: yes, the problem is that .insert() has a Entity param
use .toEntity(...) from mapper

**Review by danielursu — APPROVED** [review]

**Pull request merged**